source/Halibut/Transport/SecureListener.cs

```diff
@@ -118,49 +115,74 @@ public void Disconnect(string thumbprint)
 
         void Accept()
         {
-            var numberOfFailedAttemptsInRow = 0;
-
-            while (!cts.IsCancellationRequested)
-            {
-                try
+            // See: https://github.com/OctopusDeploy/Issues/issues/6035
+            // See: https://github.com/dotnet/corefx/issues/26034
+            var tokenCancellationListener = IsWindows()
+                ? cts.Token.Register(listener.Stop)
+                : (IDisposable) new NoopDisposable();
+            
+            var shouldAbort = IsWindows()
+                ? () => false
+                : new Func<bool>(() =>
                 {
                     SpinWait.SpinUntil(() => cts.IsCancellationRequested || listener.Pending());
+                    return cts.IsCancellationRequested;
+                });
 
-                    if (cts.IsCancellationRequested)
+            const int errorThreshold = 3;
+
+            using (tokenCancellationListener)
+            {
+                var numberOfFailedAttemptsInRow = 0;
+                while (!cts.IsCancellationRequested)
+                {
+                    try
                     {
-                        continue;
-                    }
+                        if (shouldAbort())
```

Comment:
I think if you write it similar to below, it clarifies what this part of the code does
```
void WaitForPendingConnectionOrCancellation() => SpinWait.SpinUntil(() => cts.IsCancellationRequested || listener.Pending())

// snip

if (!IsWindows())
   WaitForPendingConnectionOrCancellation();

if (cts.IsCancellationRequested)
    return;
```

```